crates/gossip/src/respond/terminate.rs

```diff
@@ -38,14 +38,18 @@ impl K2Gossip {
         };
 
         let terminated_accepted_session = if !terminated_initiated_session {
-            let read_guard = self.accepted_round_states.write().await;
+            let read_guard = self.accepted_round_states.read().await;
```

Comment:
There was a deadlock condition here

Reply:
It is usually the other way round. The acceptor would terminate, not the the initiator, so we just weren't running into this. So it's not super serious right now